### PR TITLE
Use const on accessors and unmodified string args

### PR DESCRIPTION
**Pull request by douglasgscofield**
This places const on accessors and read-only string references.  I also did a little consistency fixup on return vals for a few accessors.

### CODE REVIEW

File: src/Variant.h [hunk 1]
```diff
@@ -77,42 +78,47 @@ class VariantCallFile {
...
+            return openTabix(fn);
         } else {
-            return openFile(filename);
+            return openFile(fn);
         }
     }
 
-    bool openFile(string& filename) {
+    bool openFile(string& fn) {
+        filename = fn;
```
Comment (miguelbernadi) [inline src/Variant.h]: Unnecessary as is not used in the rest of the function

**Comment by tseemann** [comment]
This would probably help the compiler optimize. Could test in https://godbolt.org

**Comment by pjotrp** [comment]
I love this type of contribution. @douglasgscofield do you still feel like pursuing this and making sure it works?

**Comment by pjotrp** [comment]
The source tree changed to much to merge this PR easily. Point taken using 'const' more often!

**Pull request closed without merging**